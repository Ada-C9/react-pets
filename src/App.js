import React, { Component } from 'react';
// import Pet from './components/Pet'
import PetCollection from './components/PetCollection'
import axios from 'axios';
import './App.css';

// const PET_LIST =[
//   {
//     name: 'Bogart',
//     age: 42,
//     breed: 'Mixed',
//     about: 'Best cat of all times'
//   },
//   {
//     name: 'Bagel',
//     age: 1,
//     breed: 'Beagle',
//     about: 'Bagel the Beagle'
//   }
// ];


class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className="foo">Hello, World</h1>
        <PetCollection />
      </div>
    );
  }
}

export default App;
