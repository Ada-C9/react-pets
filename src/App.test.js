import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('will match the last snapshot', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });

  test('that it renders App with shallow rendering', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
