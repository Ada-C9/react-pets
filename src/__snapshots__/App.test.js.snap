// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App that it renders App with shallow rendering 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Pets!
</h1>,
        <p>
          Welcome to our pets collection. Take a look around to meet our favorite furry friends. 
</p>,
        <PetCollection
          pets={
                    Array [
                              Object {
                                "about": "",
                                "age": 12,
                                "breed": "Pug",
                                "name": "Horace",
                              },
                              Object {
                                "about": "",
                                "age": 4,
                                "breed": "Springer Spaniel",
                                "name": "Stanley",
                              },
                            ]
          }
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pets!",
        },
        "ref": null,
        "rendered": "Pets!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to our pets collection. Take a look around to meet our favorite furry friends. ",
        },
        "ref": null,
        "rendered": "Welcome to our pets collection. Take a look around to meet our favorite furry friends. ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "pets": Array [
            Object {
              "about": "",
              "age": 12,
              "breed": "Pug",
              "name": "Horace",
            },
            Object {
              "about": "",
              "age": 4,
              "breed": "Springer Spaniel",
              "name": "Stanley",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Pets!
</h1>,
          <p>
            Welcome to our pets collection. Take a look around to meet our favorite furry friends. 
</p>,
          <PetCollection
            pets={
                        Array [
                                    Object {
                                      "about": "",
                                      "age": 12,
                                      "breed": "Pug",
                                      "name": "Horace",
                                    },
                                    Object {
                                      "about": "",
                                      "age": 4,
                                      "breed": "Springer Spaniel",
                                      "name": "Stanley",
                                    },
                                  ]
            }
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pets!",
          },
          "ref": null,
          "rendered": "Pets!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to our pets collection. Take a look around to meet our favorite furry friends. ",
          },
          "ref": null,
          "rendered": "Welcome to our pets collection. Take a look around to meet our favorite furry friends. ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "pets": Array [
              Object {
                "about": "",
                "age": 12,
                "breed": "Pug",
                "name": "Horace",
              },
              Object {
                "about": "",
                "age": 4,
                "breed": "Springer Spaniel",
                "name": "Stanley",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App will match the last snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Pets!
  </h1>
  <p>
    Welcome to our pets collection. Take a look around to meet our favorite furry friends. 
  </p>
  <section
    className="pets"
  >
    <table>
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <th>
            Breed
          </th>
          <th>
            Age
          </th>
          <th>
            About
          </th>
        </tr>
        <tr
          className="pet"
        >
          <td>
            Horace
          </td>
          <td>
            Pug
          </td>
          <td>
            12
          </td>
          <td>
            
          </td>
        </tr>
        <tr
          className="pet"
        >
          <td>
            Stanley
          </td>
          <td>
            Springer Spaniel
          </td>
          <td>
            4
          </td>
          <td>
            
          </td>
        </tr>
      </tbody>
    </table>
    <form
      id="new-pet"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="name"
        >
          Pet Name
        </label>
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="breed"
        >
          Breed
        </label>
        <input
          name="breed"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="age"
        >
          Age
        </label>
        <input
          name="age"
          onChange={[Function]}
          pattern="[0-9]*"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="about"
        >
          About
        </label>
        <input
          name="about"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <input
        type="submit"
        value="Add Pet"
      />
    </form>
  </section>
</div>
`;
